firmware/src/hal/teensy/link_teensy.cpp

```
@@ -0,0 +1,184 @@
+  while (count < capacity && midi_head_ != midi_tail_) {
+    out[count++] = midi_ring_[midi_head_];
+    midi_head_ = (midi_head_ + 1) % (kMidiInputCapacity + 1);
+  }
+  return count;
+}
+
+// One byte at a time, gated on the UART having room so it never blocks — a spin here
+// would hang with interrupts off, since the caller runs the pump outside the lock but the
+// clock's own send is on the timer (D-114).
+int midi_send(const uint8_t* bytes, int count) {
+  if (count <= 0 || Serial1.availableForWrite() <= 0) return 0;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)